### PR TITLE
Namespace v1 currency code

### PR DESCRIPTION
**Pull request by aeolianeth**
V1 contracts define `ETH=0`, and `USD=1`. V2 contracts define `ETH=1` and `USD=2`. There are multiple ways to approach this, but I feel the simplest and lowest-risk way is to define new constants for the V2 currencies.

Source: https://github.com/jbx-protocol/juice-contracts-v2/blob/main/contracts/libraries/JBCurrencies.sol#L4

- Add `V1_` prefix to currency constants
- move currency files into v1 directories
- add `V1` to funding cycle types, serializers and remaining redux code
- Rename `ContractLoader` to `V1ContractLoader`

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](../CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](../CONTRIBUTING.md#browser-support).

### CODE REVIEW

File: src/components/v1/V1Project/modals/ConfirmPayOwnerModal/index.tsx [hunk 1]
```diff
@@ -104,8 +105,8 @@ export default function ConfirmPayOwnerModal({
 
         <Descriptions column={1} bordered>
           <Descriptions.Item label={t`Pay amount`} className="content-right">
-            {formattedNum(usdAmount)} {currencyName(1)} ({formatWad(weiAmount)}{' '}
-            {currencyName(0)})
+            {formattedNum(usdAmount)} {currencyName(V1_CURRENCY_USD)} (
```
Comment (aeolianeth) [inline src/components/v1/V1Project/modals/ConfirmPayOwnerModal/index.tsx:108]: Add missing constants

File: src/hooks/AppSelector.ts [hunk 2]
```diff
@@ -1,18 +1,18 @@
 import { shallowEqual, TypedUseSelectorHook, useSelector } from 'react-redux'
 import { RootState } from 'redux/store'
-import { deserializeFundingCycle } from 'utils/serializers'
+import { deserializeV1FundingCycle } from 'utils/v1/serializers'
 import { useMemo } from 'react'
 
 export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector
 
-export const useEditingFundingCycleSelector = () => {
+export const useEditingV1FundingCycleSelector = () => {
```
Comment (aeolianeth) [inline src/hooks/AppSelector.ts:8]: Add `V1` to this function name, we'll have similar v2 versions

**Comment by aeolianeth** [comment]
@johnnyd-eth @peripheralist this PR picks some changes from https://github.com/jbx-protocol/juice-interface/pull/551 in an attempt to make it a smaller PR.

Let me know what you think 🙇

**Review by johnnyd-eth — APPROVED** [review]
Yep no problems with this one @aeolianeth, nice work. Approving

File: src/components/Navbar/EthPrice.tsx [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
+import { V1_CURRENCY_ETH, V1_CURRENCY_USD } from 'constants/v1/currency'
 
 export default function EthPrice() {
   const price = useEtherPrice()
   return (
     <div>
-      <CurrencySymbol currency={CURRENCY_USD} />
-      {price?.toFixed(2)}/{currencyName(0)}
+      <CurrencySymbol currency={V1_CURRENCY_USD} />
+      {price?.toFixed(2)}/{currencyName(V1_CURRENCY_ETH)}
```
Comment (aeolianeth) [inline src/components/Navbar/EthPrice.tsx:13]: missing constant

File: src/components/shared/InputAccessoryButton.tsx [hunk 4]
```diff
@@ -19,6 +19,7 @@ export default function InputAccessoryButton({
 
   return content ? (
     <div
+      role="button"
```
Comment (aeolianeth) [inline src/components/shared/InputAccessoryButton.tsx:22]: passing improvment

File: src/components/shared/formItems/ProjectTarget.tsx [hunk 5]
```diff
@@ -23,15 +22,14 @@ export default function ProjectTarget({
...
-  onCurrencyChange: (val: CurrencyOption) => void
+  currency: V1CurrencyOption
+  onCurrencyChange: (val: V1CurrencyOption) => void
   disabled?: boolean
   fee?: BigNumber
 } & FormItemExt) {
   return (
     <Form.Item
       extra="The maximum amount of funds that can be distributed from this project in one funding cycle. Funds will be withdrawn in ETH no matter the currency you choose."
-      name={name}
```
Comment (aeolianeth) [inline src/components/shared/formItems/ProjectTarget.tsx:34]: we don't need `name` as we already have other sufficient event handlers.

**Comment by aeolianeth** [comment]
@johnnyd-eth thanks for the review! Tested on Fleek and everything appears unaffected. Merging.

**Pull request merged**